File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hm;
        vector<int> sol;
        for(int i = 0; i < nums.size(); i++){
            if(hm.find(nums[i]) != hm.end()){
                if(i < hm[nums[i]]){
                    sol.push_back(i);
                    sol.push_back(hm[nums[i]]);
                } else {
                    sol.push_back(hm[nums[i]]);
                    sol.push_back(i);
                }
            } else {
                hm[target-nums[i]] = i;
            }
        }
        return sol;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int lp = 0, rp = numbers.Length - 1;
        while(lp<= rp){
            if(numbers[lp] + numbers[rp] > target){
                rp--;
                continue;
            }
            if(numbers[lp] + numbers[rp] < target){
                lp++;
                continue;
            }
            return [lp+1,rp+1];
        }
        return [];
    }
}
